src/rabbit_auth_backend_http.erl

```diff
@@ -70,31 +70,41 @@ check_resource_access(#auth_user{username = Username},
 %%--------------------------------------------------------------------
 
 bool_req(PathName, Props) ->
-    case http_get(q(PathName, Props)) of
+    case http_req(p(PathName), q(Props)) of
         "deny"  -> false;
         "allow" -> true;
         E       -> E
     end.
 
-http_get(Path) -> http_get(Path, ?RETRY_ON_KEEPALIVE_CLOSED).
+http_req(Path, Query) -> http_req(Path, Query, ?RETRY_ON_KEEPALIVE_CLOSED).
 
-http_get(Path, Retry) ->
-    case http_get_req(Path) of
+http_req(Path, Query, Retry) ->
+    case do_http_req(Path, Query) of
         {error, socket_closed_remotely} ->
             %% HTTP keepalive connection can no longer be used. Retry the request.
             case Retry > 0 of
-                true  -> http_get(Path, Retry - 1);
+                true  -> http_req(Path, Query, Retry - 1);
                 false -> {error, socket_closed_remotely}
             end;
         Other -> Other
     end.
 
-http_get_req(Path) ->
-    URI = uri_parser:parse(Path, [{port, 80}]),
+
+do_http_req(PathName, Query) ->
+    URI = uri_parser:parse(PathName, [{port, 80}]),
     {host, Host} = lists:keyfind(host, 1, URI),
     {port, Port} = lists:keyfind(port, 1, URI),
     HostHdr = rabbit_misc:format("~s:~b", [Host, Port]),
-    case httpc:request(get, {Path, [{"Host", HostHdr}]}, [], []) of
+    {ok, Method} = application:get_env(rabbitmq_auth_backend_http, method),
```

Comment:
This name is too generic. It should be `http_method`, or even `{use_case}_http_method` because using a single method for every operation is exactly the issue in #7.